### ISSUE TITLE
Build error [ambiguous instructions require an explicit suffix] with LLVM/icx (tests/benchmarks/timing_info)

### ISSUE THREAD

**Issue by nashif** [zephyriot/zep-jira14#1999]
**_Reported by rahul kumar:_**

Test case with build errors raised from LLVM/icx: tests/benchmarks/timing_info

```
/home/Vikram/Project/zephyr/tests/benchmarks/timing_info/src/thread_bench.c:66:27: error: 
      ambiguous instructions require an explicit suffix (could be 'cmpb',
      'cmpw', 'cmpl', or 'cmpq')
                "mov %edx,start_time+4\n\t"
                                        ^
<inline asm>:6:2: note: instantiated into assembly here
        cmp $0x1,__read_swap_end_tsc_value_test
        ^
1 error generated.

```

How to setup llvm/icx
# update the icx compiler with 0214 build from ISSM team
#  source <path_to_llvm>/linux_prod/linux/bin/xcompilervars_arch.sh ia32 mcu
#  cd zephyr
#  source zephyr-env.sh
#  export ISSM_INSTALLATION_PATH=<path_to_issm_toolchain>/issm_2016.1.057
#  export ZEPHYR_GCC_VARIANT=issm

How-to-reproduce:
# cd tests/benchmarks/timing_info
# make pristine
# make BOARD=quark_se_c1000_devboard CC=icx

Expected result : Build should pass without any warning or error.

Actual result : Observed build failed due to errors.

Attached captured compilation log.

(Imported from Jira ZEP-2160)

**Comment by nashif** [comment]
**_by rahul kumar:_**

* [tests_benchmarks_timing_info](https://s3.amazonaws.com/logs.zephyrproject.org/jira/ZEP-2160/tests_benchmarks_timing_info) (size: 5581)

**Comment by nashif** [comment]
**_by Sharron LIU:_**

merged to v1.8-rc2

**Comment by nashif** [comment]
**_by vikram Shekhawat:_**

Fixed in latest v1.8rc2.

**Comment by nashif** [comment]
**_by Sharron LIU:_**

observed again at v1.8-rc3

**Comment by nashif** [comment]
**_by Adithya Baglody:_**

_Sharron LIU_  _vikram Shekhawat_  
can you please check once more.
my Commit id is bee2b84
this was done after the tag was applied. Hence the fix might not be visible in 1.8.0-rc3

**Comment by nashif** [comment]
**_by ashish sinha:_**

Fixed: Issue cannot reproduce in master :(commit ID- 0f0c45ce)
